core/imodel-bridge/src/BADGERSIssueReporter.ts

```diff
@@ -0,0 +1,288 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+import { DbResult, Guid, GuidString } from "@bentley/bentleyjs-core";
+import { ECDb } from "@bentley/imodeljs-backend";
+import { BridgeIssueReporter } from "./BridgeIssueReporter"
+import * as fs from "fs";
+import * as path from "path";
+import * as os from "os";
+import { fips } from "crypto";
+
+interface ContextInfo {
+    activityId: string;
+    contextId: string;
+    iModelId: string;
+    briefcaseId: number;
+    operationType?: string;
+    jobId?: string;
+}
+
+interface FileInfo {
+    fileId?: string;
+    filePath?: string;
+}
+
+/** BADGERS issue reporter. Can be provided to BridgeRunner in setIssueReporter
+ * @beta
+ */
+export class BADGERSIssueReporter implements BridgeIssueReporter {
+    private _reportDb;
+    private _reportFile;
+    private _reportSummaryFile; //ToDo create summary report
+    private _stagingDir;
+    private _contextInfo: ContextInfo;
+    private _fileInfo: FileInfo;
+    private _dbPath: string = "";
+    // private _authHandler; //Callback to get token. Refer to serverArgs
+    // private _signInMgr;
+    // private _isValidContext;
+    private _deleteBadgersDb;
+    private _categoryCaluse;
+    private _typeCaluse;
+    private _levelCaluse;
+
+    private TEMP_REPORT_FILE_RECORDS_TABLE = "BadgersReportFileRecords";
+    private TEMP_REPORT_AUDIT_RECORDS_TABLE = "BadgersReportAuditRecords";
+    private SRCFILE_TEMP_BRIDGE_ISSUES_TABLE = "BadgersSrcFileReports";
+    private TEMP_IGNORED_ELEM_RECORD_TABLE = "BadgersReportIgnoredElementRecords";
+
+    public constructor(contextId: any, iModelId: GuidString, activityId: string, fileId: string, operationType?: string, filePath?: any,
+        stagingDir?: any, jobId?: string, deleteReportDB?: boolean) {
+
+        const opType = operationType ? operationType : "NORMAL_UPDATE";
+        this._contextInfo = { operationType: opType, activityId, jobId, contextId, iModelId, briefcaseId: 2 };
+        this._fileInfo = { fileId, filePath };
+        this._stagingDir = stagingDir ? stagingDir : os.tmpdir();
+        this._deleteBadgersDb = deleteReportDB ? deleteReportDB : false;
```

Comment:
I'd avoid nullish coalescing... `this._deleteBadgersDb = deleteReportDB ?? false;`